tutorials/plugins/gdnative/gdnative-cpp-scons.rst

```diff
@@ -0,0 +1,872 @@
+    library_file = lib_name + "." + platform + "." + str(bits)
+
+    lib_path = ARGUMENTS.get("lib", "bin/")
+
+    lib_path = lib_path.split("\\").join("/")
+    if lib_path[-1:] != "/":
+        lib_path += "/"
+
+    # ------
+    # While we've prepared our lib_path and library_file values now,
+    # we won't create the final_lib_path value until after we've
+    # done our OS-specific logic down below, since we need to know
+    # the name of the <platform><bits> subdirectory.
+    # ------
+
+    # library dependencies
+    
+    cpp_bindings_library_path = ARGUMENTS.get("cpp_bindings_library", cpp_bindings_library + "bin/godot-cpp")
+
+    ext = "z"
+    while ext != "":
+        cpp_bindings_library_path, ext = os.path.splitext(cpp_bindings_library_path)
+
+    cpp_bindings_library = cpp_bindings_library_path + "." + platform + "." + str(bits)
+
+    other_libs = ARGUMENTS.get("other_libs", "")
+    other_libs = other_libs.split(",")
+
+    libs = [cpp_bindings_library] + other_libs
+
+    # source dependencies
+
+    godot_headers_path = ARGUMENTS.get("headers", "godot_headers/")
+    cpp_bindings_path = ARGUMENTS.get("cpp_bindings_path", "godot-cpp/")
+    sources = ARGUMENTS.get("sources", "")
+
+    source_files = []
+
+    source_dirs = sources.split(",")
+
+    for path in source_dirs:
+        if os.path.isdir(path):
+            add_sources(source_files, path)
+
+    source_dirs.append(godot_headers_path)
+    source_dirs.append(cpp_bindings_path + "include/")
+    source_dirs.append(cpp_bindings_path + "include/core/")
+
+    # OS-specific logic for flags and the output lib directory
+
+    platform_dir = ""
+
+    if platform == "osx":
+        env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64'])
+        env.Append(LINKFLAGS = ['-arch', 'x86_64'])
+
+        platform_dir = "osx"
+    
+    elif platform == "linux":
+        env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++14'])
+
+        platform_dir = "x11"
+    
+    elif platform == "windows":
+
+        # Set exception handling model to avoid warnings caused by Windows system headers.
+        env.Append(CCFLAGS=['-EHsc'])
+
+        if target == "debug":
+            env.Append(CCFLAGS = ['-D_DEBUG', '-MDd'])
+        else:
+            env.Append(CCFLAGS = ['-O2', '-DNDEBUG', '-MD'])
+
+        platform_dir = "win"
+
+    else:
+        # do nothing if we don't recognize the platform
+        print 'unrecognized platform provided. Please enter a valid platform.'
+        return
+
+    final_lib_path = lib_path + platform_dir + str(bits) + "/" + lib_name
+
+    env.Append(LIBS=libs)
+    env.Append(CPPPATH=source_dirs)
+    
+    library = env.SharedLibrary(target=final_lib_path, source=source_files)
+    Default(library)
+
+I know there's a lot to see here, and in some cases, this may be all you need. However, there are some other specific cases of usability that we need to look out for, and which could stand to improve this build file.
+
+Special Operations - LLVM
```

Comment:
Instead of "Special Operations" could we change it to "Changing the Default Compiler"